### ISSUE TITLE
Can I use Beyond IoU as a loss and metric for my own training?

### ISSUE THREAD

**Issue by JulienMaille** [mxbh/beyond-iou#1]

**Comment by mxbh** [comment]
Hi,
unfortunately, our error metrics are not differentiable, so you cannot use them as a loss function during training.

**Comment by JulienMaille** [comment]
Thank you, would you have any advice on how to port the metric to "vanilla" pyTorch?

**Comment by mxbh** [comment]
Most of our code is plain numpy, so transferring these parts to torch should be relatively easy. However, we use the function `scipy.ndimage.label`, which may be hard to realize in torch. If you want to port to torch for differentiability, major conceptual changes are necessary as our method uses the argmaxed, hard predictions of the model.
